tests/vmipreset_test.go

```diff
@@ -414,6 +414,81 @@ var _ = Describe("[rfe_id:609][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 		})
 	})
 
+	Context("[rfe_id:613]MatchLabels", func() {
+		var preset *v1.VirtualMachineInstancePreset
+		labelKey := "kubevirt.io/cpu"
+		labelValue := "dodecacore"
+		numCores := uint32(12)
+		presetName := "twelve-cores"
+
+		var annotationLabel string
+		var annotationVal string
+
+		var vmiWin7 *v1.VirtualMachineInstance
+		var vmiWin10 *v1.VirtualMachineInstance
+
+		BeforeEach(func() {
+			selector := k8smetav1.LabelSelector{
+				MatchLabels: map[string]string{
+					labelKey: labelValue,
+				},
+			}
+
+			preset = &v1.VirtualMachineInstancePreset{
+				ObjectMeta: k8smetav1.ObjectMeta{Name: presetName},
+				Spec: v1.VirtualMachineInstancePresetSpec{
+					Selector: selector,
+					Domain: &v1.DomainSpec{
+						CPU: &v1.CPU{Cores: numCores},
+					},
+				},
+			}
+
+			// The actual type of machine is unimportant here. This test is about the label
+			vmiWin7 = tests.NewRandomVMIWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskAlpine))
+			vmiWin7.Labels = map[string]string{labelKey: labelValue}
+			vmiWin10 = tests.NewRandomVMIWithEphemeralDisk(tests.ContainerDiskFor(tests.ContainerDiskAlpine))
+			vmiWin10.Labels = map[string]string{labelKey: labelValue}
+
+			annotationLabel = fmt.Sprintf("virtualmachinepreset.kubevirt.io/%s", presetName)
+			annotationVal = v1.GroupVersion.String()
+		})
+
+		It("[test_id:672] Should match multiple VMs via MatchLabel", func() {
+			By("Creating preset with MatchExpression")
+			err := virtClient.RestClient().Post().Resource("virtualmachineinstancepresets").Namespace(tests.NamespaceTestDefault).Body(preset).Do().Error()
```

Comment:
for new code, we should start using the api wrappers we already have instead of the RestClient directly

Reply:
My bad. I was of the impression we didn't have api wrappers for presets. Thanks for pointing that out.

Reply:
@davidvossel it turns out that the api wrappers for presets were indeed broken.

Once this PR

https://github.com/kubevirt/kubevirt/pull/2299

Is merged, I will rebase and fix this test.